services/publisher/publisher.go

```diff
@@ -155,6 +155,16 @@ func (pub *Publisher) publishNextBatch(batch *protocol.AlertBatch) (published bo
 		}
 	}
 
+	// add chain id if it's not set, e.g. EventMetric
+	// TODO: based on the bot id query agent config and use chain id from it or do that inside the aggregator
+	for _, metric := range batch.Metrics {
```

Comment:
Shall we move this loop into the aggregator, inside `PrepareMetrics()`?

Reply:
The aggregator can be provided the scanner chain ID at initialization. (as discussed in the Huddle)